src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,27 @@
 package mate.academy;
 
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Random random = new Random();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
```

Comment:
The `processRequest` method should check if the `userId` is already in the cache before proceeding with processing. If it is, return the cached `UserData` immediately.